astropy/units/equivalencies.py

```diff
@@ -371,40 +380,21 @@ def doppler_radio(rest):
     <Quantity -31.209092088877583 km / s>
     """
     assert_is_spectral_unit(rest)
-
-    ckms = _si.c.to_value("km/s")
-
-    def to_vel_freq(x):
-        restfreq = rest.to_value(si.Hz, equivalencies=spectral())
-        return (restfreq - x) / (restfreq) * ckms
-
-    def from_vel_freq(x):
-        restfreq = rest.to_value(si.Hz, equivalencies=spectral())
-        voverc = x / ckms
-        return restfreq * (1 - voverc)
-
-    def to_vel_wav(x):
-        restwav = rest.to_value(si.AA, spectral())
-        return (x - restwav) / (x) * ckms
-
-    def from_vel_wav(x):
-        restwav = rest.to_value(si.AA, spectral())
-        return restwav * ckms / (ckms - x)
-
-    def to_vel_en(x):
-        resten = rest.to_value(misc.eV, equivalencies=spectral())
-        return (resten - x) / (resten) * ckms
-
-    def from_vel_en(x):
-        resten = rest.to_value(misc.eV, equivalencies=spectral())
-        voverc = x / ckms
-        return resten * (1 - voverc)
-
+    spectral_eq = functools.partial(rest.to_value, equivalencies=spectral())
+    to_funcs = {
+        si.Hz: lambda x: (1 - x / spectral_eq(si.Hz)) * ckms,
```

Comment:
```suggestion
        si.Hz: lambda x: (1 - x / spectral_eq(si.Hz)) << ckms,
```

Does this speed it up further?

Reply:
`ckms` is not a unit!

Reply:
Then it should not speed it up!

Reply:
Indeed, it should give an error, since one cannot left shift with a float...